File: WEEKLYCODE#2/sherlock_and_array/Java/sherlock_and_array.java
import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

public class Solution
{
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        int t = input.nextInt();
        //for each test case
            for(int j=0;j<t;j++) {
                int n = input.nextInt();
                int a[] = new int[n];
                long total = 0;
                //taking input and calulating total sum
                for (int i = 0; i < n; i++) {
                    a[i] = input.nextInt();
                    total += a[i];
                }
                //checking for element
                int sumLeft = 0;
                String result = "NO";
                for (int i = 0; i < n; i++) {
                    if (total - sumLeft - a[i] == sumLeft) {
                        result = "YES";
                        break;
                    }
                    sumLeft += a[i];
                }
                //printing result
                System.out.println(result);
            }
    }
}

File: WEEKLYCODE#2/sherlock_and_array/Python/sherlock_and_array.py
#for each test case
for t in xrange(input()):
    #taking input
    n = input()
    a = map(int, raw_input().split())
    x = 0
    #total sum
    y = sum(a)
    ans = False
    #checking for element
    for i in xrange(n):
        y -= a[i]
        if x == y:
            ans = True
        x += a[i]
    #printing result
    print 'YES' if ans else 'NO'